### DISCUSSION TITLE
FW upgrades

### DISCUSSION THREAD

**Discussion by jorgensigvardsson** [westermo/netbox#37] [Ideas]
It would be really awesome if FW upgrades could support "transactional" semantics. With transactional I mean that the agent (tool or human) upgrading the device gets to initiate the upgrade, and then commit or abort. Upon commit, it's super important to get an "ack" back before the device reboots. This would make life a lot easier for NCM tools (humans too I suspect).

It would be even nicer if the FW upgrade process could emit events in some form, indicating progress. Percent progress is nice, but hard to achieve. Stages, or any other "life signs" would be a good indication for tools _and_ humans.

Thoughts?

**Comment by Hexxel** [comment]
It would seem that this problem has been solved by others. Any tools/patterns suggestions? fwupd?

**Reply by troglobit** [reply]
Agreed, there's also https://www.rauc.io/ which is extremely cool.

Also, this is probably something we want for projects that build *on top of* NetBox.  For NetBox itself I'd recommend keeping it simple and not invest too much.  The value-add from Westermo products built on NetBox should be clear in every stage IMO

**Reply by jorgensigvardsson** [reply]
I had forgotten about RAUC. It is a very cool project! It is however not clear (from glancing the documentation) how a manual/forced OTA update should work. Ideally, in an industrial production network, you probably don't want fully autonomous upgrades. You'd probably want to set a date when an NMS/NCM may push new updates, or maybe even manually by an operator that observes the process. It could also mean that all the low level talk zips over my ignorant head a friday evening... :)

**Reply by Hexxel** [reply]
Do we have any requirements on upgrade? Are we free to come up with a new (better(!)) way of doing it? If so a wishlist is a good start...
